components/admin/events/new-dinner-dialog.tsx

```
@@ -0,0 +1,270 @@
+		key: K,
+		value: FormFields[K],
+	) => {
+		setFields((prev) => ({ ...prev, [key]: value }));
+	};
+
+	const handleClose = () => {
+		setFields(INITIAL_FIELDS);
+		onClose();
+	};
+
+	const handleSubmit = async () => {
```

Biome (2.4.4)

[error] Excessive complexity of 15 detected (max: 12).

(lint/complexity/noExcessiveCognitiveComplexity)